includes/class-wc-gateway-komoju-webhook-event.php

```diff
@@ -0,0 +1,121 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * An encapsulation around the data from Webhook events. This allows us to
+ * change the structure of the webhook request data without having to change it
+ * throughout the code.
+ */
+class WC_Gateway_Komoju_Webhook_Event {
+
+    private $requestJson;
+
+    /**
+	 * Constructor
+	 * @param string $requestBody the body of the webhook request
+	 */
+    public function __construct($requestBody) {
+        $this->requestJson = json_decode($requestBody, true);
+
+        if (! empty( json_last_error() )) {
+            $errorMsg = "Komoju IPN Request JSON Decoding Failure. Error: " . json_last_error_msg();
+            wp_die( $errorMsg, "Komoju IPN", array( 'response' => 500 ) );
```

Comment:
If we can't decode the webhook request as valid JSON then we can't do anything with it, so I figured it was better to kill it with a non success code, so we can see it come up as failed in the Komoju console. 

But as I'm looking at it now, I think a 400 would be better, since if the code is dying here then it's due to an issue with the request (AFAIK)

Reply:
Switched to 400 in b4658f1.